### PR TITLE
Fix build issues on non-linux platforms

### LINKED ISSUE #11676: Fix build issues on non-linux platforms
Try fix the build issue on non-linux platform.

What's Changed: components/resource_metering/Cargo.toml

### PR DESCRIPTION
**Pull request by devillove084**
### What is changed and how it works?

Try fix the build issue on non-linux platform.
Close #11676

What's Changed: components/resource_metering/Cargo.toml

### Related changes

- None

```release-note
NONE
```

### CODE REVIEW

File: components/resource_metering/src/utils.rs [hunk 1]
```diff
@@ -3,6 +3,7 @@
 #![allow(dead_code)]
 
 use collections::HashSet;
+use std::thread;
```
Comment (sticnarf) [inline components/resource_metering/src/utils.rs:6]: It causes a warning if the build platform is linux. I suggest you write `std::thread::current().id().as_u64().get() as usize` in the `thread_id` function.

**Review by sticnarf — COMMENTED** [review]
It looks good to me.

Please sign off your commit to pass the DCO check.

And PTAL @zhongzc

**Comment by zhongzc** [comment]
Just need to add the `thread-id` dependency to `components/resource_metering/Cargo.toml`. It was deleted by accident recently.

```
[target.'cfg(not(target_os = "linux"))'.dependencies]
thread-id = "4"
```

File: components/resource_metering/src/utils.rs (hunk 1 above)
Comment (devillove084) [reply components/resource_metering/src/utils.rs]: done

**Comment by devillove084** [comment]
cc @zhongzc

File: components/resource_metering/src/lib.rs [hunk 2]
```diff
@@ -5,6 +5,7 @@
 #![feature(shrink_to)]
 #![feature(hash_drain_filter)]
 #![feature(core_intrinsics)]
+#![feature(thread_id_value)]
```
Comment (zhongzc) [inline components/resource_metering/src/lib.rs:8]: Please remove this line

**Review by zhongzc — APPROVED** [review]
rest LGTM

File: components/resource_metering/src/lib.rs (hunk 2 above)
Comment (devillove084) [reply components/resource_metering/src/lib.rs]: done

**Review by andylokandy — APPROVED** [review]

**Comment by zhongzc** [comment]
@devillove084  Please follow these guides to resolve all `do-not-merge` labels:

> Adding the "do-not-merge/release-note-label-needed" label because no release-note block was detected, please follow our release note process to remove it.

> Please provide the associated issue number in the commit messages, for example: close #12345, or ref #12345.

**Pull request merged**